### DISCUSSION TITLE
How to configure the resoure dictionary

### DISCUSSION THREAD

**Discussion by blackwhite-bear** [DioxusLabs/dioxus#442] [General]
- Here is the code I wrote, but it is not able to be compiled
- The reason I want to reset the path of resource dictionary is that it is better to looks like a single bundled application running on rather than generating a lot of files aside
- Every elements in the window can be selected. It is the best to allow configured elements able to selected

``` rust
use std::path::PathBuf;
use dioxus::prelude::*;

fn main() {
    dioxus::desktop::launch_cfg(win_elements, |app| {
        let mut resource_path = PathBuf::new();
        resource_path.push(r"C:\");
        resource_path.push("tmp");
        app.with_disable_context_menu(true)
            .with_custom_head(String::from("ABC"))
            .with_window(|w|{
                w.with_title("title")
            })
            .with_resource_directory(resource_path.into())
    });
}

fn win_elements(cx: Scope) -> Element {
    cx.render(rsx! {
        div {
            "hello world!"
        }
    })
}
```

**Comment by rMazeiks** [comment]
Yeah, it appears that there's an inconsistency in `with_resource_directory` – other methods take and return `&mut self`, but it takes `self` – which doesn't work out since the `launch_cfg` closure only provides a `&mut` reference. (And it would also not work out due to lifetimes)

I think `with_resource_directory` must be fixed – this is a bug in Dioxus. It should have the signature

```rust
pub fn with_resource_directory(&mut self, path: impl Into<PathBuf>) -> &mut Self
```

And then your code would work (though the `.into()` will be unnecessary).

**Reply by blackwhite-bear** [reply]
Good！
- hope this patch can be delivered soon
- how about the function of selected elements? is there a way to block it but configured one?
